src/modules/FtpServer.rb

```diff
@@ -443,10 +443,9 @@ def WriteFirewallSettings
       return true if !firewalld.running?
 
       if Ops.get(@EDIT_SETTINGS, "PassiveMode") == "YES"
-        port_range = Ops.add(
-          Ops.add(Ops.get(@EDIT_SETTINGS, "PasMinPort"), ":"),
-          Ops.get(@EDIT_SETTINGS, "PasMaxPort")
-        )
+        raise ArgumentError, "Missing min / max ports for passive ftp" if @EDIT_SETTINGS["PasMinPort"].nil? || @EDIT_SETTINGS["PasMaxPort"].nil?
```

Comment:
I'm fine adding the exception, although the line seems to bee tooooo long and also if `nil` is the only option to be checked probably it is enough checking if it is defined with `unless @EDIT_SETTINGS["PasMinPort"] || @EDIT_SETTINGS["PasMaxPort"]` but if not set would be `nil` or could be also an empty string which is also invalid?

Reply:
Also if we are not going to handle the exception, maybe it is preferable to just report an error and return false

Reply:
I'd stay with exception. I've added that obly bcs of my paranoia from old code. In fact everything should be checked - as it is user's input - here.